File: suites/osprey-compute-kernels/reference/cases/fib/fib.ml
(* Naive recursive Fibonacci — exercises raw function-call + recursion overhead. *)
let add a b = a + b
let sub a b = a - b

let rec fib n =
  match n with
  | 0 -> 0
  | 1 -> 1
  | _ -> add (fib (sub n 1)) (fib (sub n 2))

let () = Printf.printf "%d\n" (fib 35)

File: suites/osprey-compute-kernels/reference/cases/fib/fib.hs
-- Naive recursive Fibonacci — exercises raw function-call + recursion overhead.
add :: Int -> Int -> Int
add a b = a + b

sub :: Int -> Int -> Int
sub a b = a - b

fib :: Int -> Int
fib 0 = 0
fib 1 = 1
fib n = add (fib (sub n 1)) (fib (sub n 2))

main :: IO ()
main = print (fib 35)
